code/FsComposite/Composite.Github/Request.fs

```diff
@@ -0,0 +1,181 @@
+
+        let addJsonBody body (req : IRestRequest) =
+            req.RequestFormat <- DataFormat.Json
+            req.AddBody(body)
+        let getResponse req =
+            client.Execute(req)
+
+        let rec executeWith getResponseMethod obj2 =
+            try
+                match obj2 with
+                | Request (RequestSetInBody (o, s)) -> if s = Set.empty
+                                                       then raise (MissingRequestBodyException "Request was supposed to have a body. Execution cancelled.")
+                                                       else Request o |> executeWith (addJsonBody s >> getResponseMethod)
+                | Request req -> Response (matchRequest getResponseMethod req)
+                | x -> invalidOp (sprintf "Only GithubObject.Request supports execution but here the type %s" (GetUnionCaseName x))
+            with
+            | o -> Response (Error o)
+
+        obj |> executeWith getResponse
+
+    let rec execute client obj =
+        let try_get_next_request obj =
+            match obj with
+            | PrReadResponse resp ->
+                match get_next_page_url github_source_url resp with
+                | Some url -> Some (PrReadRequest (create_next_page_rest_request url))
+                | None -> None
+            | PrsReadResponse resp ->
+                match get_next_page_url github_source_url resp with
+                | Some url -> Some (PrsReadRequest (create_next_page_rest_request url))
+                | None -> None
+            | PrFilesReadResponse resp ->
+                match get_next_page_url github_source_url resp with
+                | Some url -> Some (PrFilesReadRequest (create_next_page_rest_request url))
+                | None -> None
+            | PrCommitsReadResponse resp ->
+                match get_next_page_url github_source_url resp with
+                | Some url -> Some (PrCommitsReadRequest (create_next_page_rest_request url))
+                | None -> None
+            | PrCommentsReadResponse resp ->
+                match get_next_page_url github_source_url resp with
+                | Some url -> Some (PrCommentsReadRequest (create_next_page_rest_request url))
+                | None -> None
+            | IssueCommentsReadResponse resp ->
+                match get_next_page_url github_source_url resp with
+                | Some url -> Some (IssueCommentsReadRequest (create_next_page_rest_request url))
+                | None -> None
+            | LabelsReadResponse resp ->
+                match get_next_page_url github_source_url resp with
+                | Some url -> Some (LabelsReadRequest (create_next_page_rest_request url))
+                | None -> None
+            | SearchCodeReadResponse resp ->
+                match get_next_page_url github_source_url resp with
+                | Some url -> Some (SearchCodeRequest (create_next_page_rest_request url))
+                | None -> None
+            | _ -> None

```

Comment:
This thing here can obviously be simplified. Consider extraction of the Some/None common logic.

Reply:
Are you talk about wrapping the function 'get_next_page_url' into another function with the fixed parameter 'github_source_url '?